pkg/cmd/pulumi/deployment.go

```diff
@@ -82,7 +85,110 @@ func newDeploymentSettingsCmd() *cobra.Command {
 		}),
 	}
 
+	cmd.AddCommand(newDeploymentSettingsInitCmd())
 	cmd.AddCommand(newDeploymentSettingsPullCmd())
+	cmd.AddCommand(newDeploymentSettingsUpdateCmd())
+	cmd.AddCommand(newDeploymentSettingsDestroyCmd())
+
+	return cmd
+}
+
+func newDeploymentSettingsInitCmd() *cobra.Command {
+	var stack string
+
+	cmd := &cobra.Command{
+		Use:        "init",
```

Comment:
What's the full command for this? `pulumi deploy settings init`?

Reply:
yes

> Does init have a wizard guide you through creation of the deployment settings? Is that in the plan? Or is it mainly guessing at the setup based on stack tags?

I am initially just guessing info from the context to keep it simple (mostly repo and branch).

Reply:
Looks like we are assuming the github app is enabled. What about bitbucket and/or gitlab repos? Those could still have deployments enabled - gitops workflows wouldn't work but api-driven and click to deploy would.

Reply:
Anything other than GitHub will support DS updates through the CLI. We wont support any gitops flow for those as it would require to clone the repo and it opens the service to many security vulnerabilities. Doing it in a secure way, will require a lot of extra effort so we decided to cut the scope on this.

For GitHub, it assumes the app is installed, once we turn it into a wizard it could ask the user for that and go in different directions.

For Non-Github, it is configuring the repo url, but it is not asking for any kind of credentials. Once we convert it into a wizard, it can ask for it (tho there will be another command to deal with credentials so users can still set those values through that command)

Reply:
Thanks for explaining, I see all that now in the code :)

Might be worth sprinkling a few comments in there to this effect so it's a little easier to grok all the decisions we're making.

Reply:
I will